### PR TITLE
Entities filtering ordering

### PR DESCRIPTION
**Pull request by kkateq**
1. Finished filter, ordering entities
2. Reduced boilerplate with the redux sagas, replaced with universal `fetchAnything` saga

### CODE REVIEW

File: src/containers/PullRequestContainer/index.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
+/* @flow */
+/* eslint-disable import/no-extraneous-dependencies */
```
Comment (skovhus) [inline src/containers/PullRequestContainer/index.js]: Which import is extraneous? Nothing should be.

File: src/containers/PullRequestContainer/index.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    this.props.dispatch(this.props.fetchData(args))
+  }
+
+  handleOrderFieldSelect = (field: string): void => {
+    const args = this.getArguments()
+    args.orderBy.field = field
+    this.props.dispatch(this.props.fetchData(args))
+  }
+
+  handleRemove = (id) => {
```
Comment (skovhus) [inline src/containers/PullRequestContainer/index.js]: I would add a FIXME to indicate that this is not done

File: src/ducks/pagination/index.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 /* @flow */
 
-export const REQUEST_PAGE = 'REQUEST_PAGE'
+// export const REQUEST_PAGE = 'REQUEST_PAGE'
```
Comment (skovhus) [inline src/ducks/pagination/index.js]: Dead code?

File: src/sagas/fetch/index.js [hunk 4]
```diff
@@ -18,4 +18,24 @@ export function* fetchSaga(
   return null
 }
 
+export function* fetchAnythingSaga(action: Object): Generator<any, any, any> {
```
Comment (skovhus) [inline src/sagas/fetch/index.js]: Let us type the `action` object.

E.g. `export function* fetchAnythingSaga(action: { name: string, query: string, args: Array<string>, callback: ?Function, ... })`

Comment (skovhus) [inline src/sagas/fetch/index.js]: Maybe the type should be defined in ducks/fetch

File: src/sagas/fetch/index.js [hunk 5]
```diff
@@ -18,4 +18,24 @@ export function* fetchSaga(
...
+  try {
+    yield put(actions.clearError(action.name))
+    yield put(actions.sendingRequest(action.name, true))
+    const response = yield call(get, action.query, action.args)
+    if (action.callback) {
+      const callbacks = action.callback(response, action.args)
+      while (callbacks.length) {
```
Comment (skovhus) [inline src/sagas/fetch/index.js]: So a `action.callback` can return a list of callback functions?

Comment (skovhus) [inline src/sagas/fetch/index.js]: Seems a bit more idiomatic to just do a map, `callbacks.map(callback => yield put(callback))`

File: src/sagas/fetch/index.js (hunk 4 above)
Comment (kkateq) [inline src/sagas/fetch/index.js]: Why? It referenced only from root saga, don't think we should mess up sagas within ducks

File: src/sagas/fetch/index.js (hunk 5 above)
Comment (kkateq) [inline src/sagas/fetch/index.js]: well, it's not possible :) generators can be called only from generators, `map` is iterator but not the generator, so the `yield` won't work with `map`

Comment (skovhus) [inline src/sagas/fetch/index.js]: Ah... Didn't know that.

**Pull request merged**